Add inject command to update collection items from .tif files and update propierties

denarch22 opened the pull request:
## 🛠️ Changes
Introduces an 'inject' command to update the 'items' section of input/<folder>/collection.json based on .tif files present in the folder, preserving other metadata. Updates Collection logic to ensure 'proj:epsg' is present for each item, computing it if missing, and adds projection extension support. Documentation and CLI argument parsing are updated to reflect the new functionality.

## 📝 Associated issues
Resolve [LIB-246](https://linear.app/biodev/issue/LIB-246/agregar-propiedad-de-proyeccion-para-items)

denarch22 committed:
Add inject command to update collection items from .tif files

Introduces an 'inject' command to update the 'items' section of input/<folder>/collection.json based on .tif files present in the folder, preserving other metadata. Updates Collection logic to ensure 'proj:epsg' is present for each item, computing it if missing, and adds projection extension support. Documentation and CLI argument parsing are updated to reflect the new functionality.

cgalvist commented on README.md:
```diff
@@ -198,6 +198,39 @@ Este comando eliminará la colección `my_collection` del sistema.
+### Ejemplos:
+
+
+### Inyectar ítems sobrescribiendo collection.json y creando backup
```
Una pequeña observación de formato. Ya que los ejemplos están debajo de la sección `### Ejemplos:`, los títulos deberían tener 4 símbolos `#` en lugar de solo 3

cgalvist reviewed (changes requested):
Hola.

No hago comentarios acerca del código en Python ni de la funcionalidad del desarrollo. Solo me he limitado a hacer pruebas siguiendo la documentación en mi entorno local.

Al ejecutar el comando para actualizar el archivo `collections.json` en mi ambiente local, con los datos de coberturas no me presenta errores, pero al hacerlo con pérdida y persistencia me sale esto:

```sh
python src/main.py inject -f lossPersistence --no-backup

2025-08-12 09:14:00,129 - INFO - Token updated successfully.
2025-08-12 09:14:00,137 - INFO - Reading base collection from: /media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/input/lossPersistence/collection.json
2025-08-12 09:14:00,138 - INFO - Found 5 .tif files in /media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/input/lossPersistence
2025-08-12 09:14:00,140 - INFO - Detected period in filename 'Colombia_pp-2000_2005_.tif': id=2000-2005, year=2005
2025-08-12 09:14:00,140 - INFO - Detected period in filename 'Colombia_pp_2000_2005.tif': id=2000-2005, year=2005
Traceback (most recent call last):
  File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/main.py", line 175, in <module>
    main()
  File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/main.py", line 161, in main
    update_collection_json_inplace(
  File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/collection.py", line 395, in update_collection_json_inplace
    raise ValueError(
ValueError: ERROR: Duplicate item id '2000-2005' derived from .tif files.
```

¿Esto sería un problema en el archivo `tif`?

**Actualización:** Olvidé mencionar que al ejecutar el comando `flake8 src`, presenta unos errores de estilo

erikasv commented on README.md:
```diff
@@ -198,6 +198,39 @@ Este comando eliminará la colección `my_collection` del sistema.
 ---
 
+## Inyectar ítems en una colección existente (inject)
+Este comando lee el archivo collection.json existente en input/<carpeta> y reemplaza su sección items usando los .tif presentes en la carpeta, manteniendo la demás información.
```
Aquí hace falta aclarar que los archivos .tiff solo pueden tener de nombre un año o un periodo.

erikasv commented on README.md:
```diff
@@ -198,6 +198,39 @@ Este comando eliminará la colección `my_collection` del sistema.
 
 ---
 
+## Inyectar ítems en una colección existente (inject)
```
No sé si solo me pasó a mi, pero de entrada no entendí el resultado del comando, tal vez sería buen arrancar por decir que el resultado del comando es un archivo `.json` que cumple la especificación para crear una colección, ya sea el mismo archivo `collection.json` o el indicado por el usuario.

Si alguien más está de acuerdo reaccione o comente aquí mismo, sino, puedes ignorar este comentario

denarch22 commented:
> Hola.
> 
> No hago comentarios acerca del código en Python ni de la funcionalidad del desarrollo. Solo me he limitado a hacer pruebas siguiendo la documentación en mi entorno local.
> 
> Al ejecutar el comando para actualizar el archivo `collections.json` en mi ambiente local, con los datos de coberturas no me presenta errores, pero al hacerlo con pérdida y persistencia me sale esto:
> 
> ```shell
> python src/main.py inject -f lossPersistence --no-backup
> 
> 2025-08-12 09:14:00,129 - INFO - Token updated successfully.
> 2025-08-12 09:14:00,137 - INFO - Reading base collection from: /media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/input/lossPersistence/collection.json
> 2025-08-12 09:14:00,138 - INFO - Found 5 .tif files in /media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/input/lossPersistence
> 2025-08-12 09:14:00,140 - INFO - Detected period in filename 'Colombia_pp-2000_2005_.tif': id=2000-2005, year=2005
> 2025-08-12 09:14:00,140 - INFO - Detected period in filename 'Colombia_pp_2000_2005.tif': id=2000-2005, year=2005
> Traceback (most recent call last):
>   File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/main.py", line 175, in <module>
>     main()
>   File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/main.py", line 161, in main
>     update_collection_json_inplace(
>   File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/collection.py", line 395, in update_collection_json_inplace
>     raise ValueError(
> ValueError: ERROR: Duplicate item id '2000-2005' derived from .tif files.
> ```
> 
> ¿Esto sería un problema en el archivo `tif`?
> 
> **Actualización:** Olvidé mencionar que al ejecutar el comando `flake8 src`, presenta unos errores de estilo

El comando inject falló porque hay más de un GeoTIFF que representa el mismo periodo 2000-2005, lo que produciría dos ítems con el mismo id. Para mantener la unicidad de IDs en STAC, el proceso se detiene.

denarch22 commented:
> Hola.
> 
> No hago comentarios acerca del código en Python ni de la funcionalidad del desarrollo. Solo me he limitado a hacer pruebas siguiendo la documentación en mi entorno local.
> 
> Al ejecutar el comando para actualizar el archivo `collections.json` en mi ambiente local, con los datos de coberturas no me presenta errores, pero al hacerlo con pérdida y persistencia me sale esto:
> 
> ```shell
> python src/main.py inject -f lossPersistence --no-backup
> 
> 2025-08-12 09:14:00,129 - INFO - Token updated successfully.
> 2025-08-12 09:14:00,137 - INFO - Reading base collection from: /media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/input/lossPersistence/collection.json
> 2025-08-12 09:14:00,138 - INFO - Found 5 .tif files in /media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/input/lossPersistence
> 2025-08-12 09:14:00,140 - INFO - Detected period in filename 'Colombia_pp-2000_2005_.tif': id=2000-2005, year=2005
> 2025-08-12 09:14:00,140 - INFO - Detected period in filename 'Colombia_pp_2000_2005.tif': id=2000-2005, year=2005
> Traceback (most recent call last):
>   File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/main.py", line 175, in <module>
>     main()
>   File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/main.py", line 161, in main
>     update_collection_json_inplace(
>   File "/media/cgt/Data/Git/PEM-Humboldt/STAC/stac-data-tools/src/collection.py", line 395, in update_collection_json_inplace
>     raise ValueError(
> ValueError: ERROR: Duplicate item id '2000-2005' derived from .tif files.
> ```
> 
> ¿Esto sería un problema en el archivo `tif`?
> 
> **Actualización:** Olvidé mencionar que al ejecutar el comando `flake8 src`, presenta unos errores de estilo

Sobre el error al ejecutar flake8 src

El primer problema se originó al modificar la configuración de .flake8 para:

Aumentar el límite de línea a max-line-length = 120.

Ignorar reglas específicas E203 y W503 para alinearnos con el estándar de black.

Con estos cambios, flake8 comenzó a ejecutarse correctamente y a detectar errores de estilo reales en el código. Sin embargo, al intentar integrarlo con otras herramientas ( black, etc.), se presentó un conflicto de dependencias:

flake8 requería pycodestyle < 2.12.0.

autopep8 requería pycodestyle >= 2.12.0.

Esto hacía que flake8 fallara antes de analizar el código.

Solución aplicada:

Actualización de flake8 a una versión compatible (7.1.1) y fijado de versiones de pycodestyle, pyflakes y mccabe para evitar conflictos.

Instalación de las herramientas de lint y formateo (flake8, black, autopep8, isort, autoflake) vía pip en lugar de Conda, para no afectar dependencias pesadas como gdal o rasterio.

Creación de un script src/format.py que ejecuta todos los pasos de limpieza y formateo en secuencia, permitiendo aplicar los ajustes de forma más rápida y consistente.

Con esto:

flake8 src ya no presenta errores de dependencias.

El formateo y la limpieza se pueden aplicar con un solo comando como se indica en el readme

denarch22 committed:
Add format script and update style/lint config

Introduces src/format.py to automate code formatting and linting using autoflake, isort, black, autopep8, and flake8. Adds .flake8 config file and updates README with instructions for style checks and formatting. Updates environment.yml to install style tools via pip. Refactors imports for consistency and minor cleanup across source files.

cgalvist commented on .idea/misc.xml:
Por favor, descartar la carpeta `.idea` con el archivo `.gitignore`

cgalvist commented on .idea/workspace.xml:
Por favor, descartar la carpeta `.idea` con el archivo `.gitignore`

cgalvist reviewed (changes requested):
Gracias por la aclaración. Luego de eliminar el geotiff adicional, el comando ya no me da errores. Sin embargo, con el último commit tengo estas obervaciones:

- Al ejecutar el comando `python src/format.py ` me aparece este error (tengo instalado autoflake en conda):

<img width="885" height="47" alt="Image" src="https://github.com/user-attachments/assets/31a58d2f-8ffd-4010-8744-2945ae47f565" />

- Al ejecutar el comando `black src --check` indica que hay un problema de formato en el archivo `storage.py`

<img width="859" height="107" alt="Image" src="https://github.com/user-attachments/assets/f398b194-1ba7-4c57-b9c6-8a2366c1fbf3" />

denarch22 commented:
> Gracias por la aclaración. Luego de eliminar el geotiff adicional, el comando ya no me da errores. Sin embargo, con el último commit tengo estas obervaciones:
> 
> - Al ejecutar el comando `python src/format.py ` me aparece este error (tengo instalado autoflake en conda):
> 
> <img width="885" height="47" alt="Image" src="https://github.com/user-attachments/assets/31a58d2f-8ffd-4010-8744-2945ae47f565" />
> 
> - Al ejecutar el comando `black src --check` indica que hay un problema de formato en el archivo `storage.py`
> 
> <img width="859" height="107" alt="Image" src="https://github.com/user-attachments/assets/f398b194-1ba7-4c57-b9c6-8a2366c1fbf3" />

Si quieres borra el ambiente actual y recrealo nuevamente ya que hay algunos inconvenientes entre la instalación de conda y pip

cgalvist commented:
> > Gracias por la aclaración. Luego de eliminar el geotiff adicional, el comando ya no me da errores. Sin embargo, con el último commit tengo estas obervaciones:
> > 
> > * Al ejecutar el comando `python src/format.py ` me aparece este error (tengo instalado autoflake en conda):
> > 
> > <img alt="Image" width="885" height="47" src="https://private-user-images.githubusercontent.com/7953261/478062582-31a58d2f-8ffd-4010-8744-2945ae47f565.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxOTQxMjksIm5iZiI6MTc1NTE5MzgyOSwicGF0aCI6Ii83OTUzMjYxLzQ3ODA2MjU4Mi0zMWE1OGQyZi04ZmZkLTQwMTAtODc0NC0yOTQ1YWU0N2Y1NjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTRUMTc1MDI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODBmOTZmMjU3MmJmMjMwYzQxYTUzNmVmMjEwNTYwNzg4MGQ5ZmIyMTJlZDBjMWYyYWE1MWQwODRmMWE1YTIwYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.IS6sjU0eC2PZ3N3RFXS8aAWb0tCsLeMWYUJhwPj2t3o">
> > 
> > * Al ejecutar el comando `black src --check` indica que hay un problema de formato en el archivo `storage.py`
> > 
> > <img alt="Image" width="859" height="107" src="https://private-user-images.githubusercontent.com/7953261/478061500-f398b194-1ba7-4c57-b9c6-8a2366c1fbf3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxOTQxMjksIm5iZiI6MTc1NTE5MzgyOSwicGF0aCI6Ii83OTUzMjYxLzQ3ODA2MTUwMC1mMzk4YjE5NC0xYmE3LTRjNTctYjljNi04YTIzNjZjMWZiZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTRUMTc1MDI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2E3NmQ1YTJiMmE3ZWU2MjdhOGVlMzI4ZjYwY2Y3ZmM4OTI1ZmYwM2RiZGIxY2YyNWVmNDhlZGUzMmZjOWNlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.kwpV_zbANdlLQsywnvd_Cmji24qVQN6jrThkoY0wytU">
> 
> Si quieres borra el ambiente actual y recrealo nuevamente ya que hay algunos inconvenientes entre la instalación de conda y pip

Ya volví a crear el ambiente y el script no falla, pero lo demás que comenté sigue igual

denarch22 commented:
> > Gracias por la aclaración. Luego de eliminar el geotiff adicional, el comando ya no me da errores. Sin embargo, con el último commit tengo estas obervaciones:
> > 
> > * Al ejecutar el comando `python src/format.py ` me aparece este error (tengo instalado autoflake en conda):
> > 
> > <img alt="Image" width="885" height="47" src="https://private-user-images.githubusercontent.com/7953261/478062582-31a58d2f-8ffd-4010-8744-2945ae47f565.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxOTMxNTAsIm5iZiI6MTc1NTE5Mjg1MCwicGF0aCI6Ii83OTUzMjYxLzQ3ODA2MjU4Mi0zMWE1OGQyZi04ZmZkLTQwMTAtODc0NC0yOTQ1YWU0N2Y1NjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTRUMTczNDEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2NiYjAxYzM1NzVlNDBhMTJjNWQ2NzhkZjViYzhlYWI0YWJkMmFiY2RmZTAzYmVkZDdhNmMzMDRhNjZlOGJkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.txsJuWE6Et-EpTljslLNwwD01vijooC6FR-VACNgGoA">
> > 
> > * Al ejecutar el comando `black src --check` indica que hay un problema de formato en el archivo `storage.py`
> > 
> > <img alt="Image" width="859" height="107" src="https://private-user-images.githubusercontent.com/7953261/478061500-f398b194-1ba7-4c57-b9c6-8a2366c1fbf3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxOTMxNTAsIm5iZiI6MTc1NTE5Mjg1MCwicGF0aCI6Ii83OTUzMjYxLzQ3ODA2MTUwMC1mMzk4YjE5NC0xYmE3LTRjNTctYjljNi04YTIzNjZjMWZiZjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTRUMTczNDEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Yjk1YThhZjhkYjg2YzNhYzg4NDVmNDdlMjAyM2UwMjYxOTg2ODEzZTRjYTI5ODA5NmUyZTVjYmNjYWU0NDhlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.56ZnQ5rztBGi5M7hHr4wqKYOm6ltiy6I1RS2Z36077c">
> 
> Si quieres borra el ambiente actual y recrealo nuevamente ya que hay algunos inconvenientes entre la instalación de conda y pip

cuando el script del formato se ejecuta correctamente, black tambien :
<img width="788" height="107" alt="image" src="https://github.com/user-attachments/assets/567318a5-c4e7-410c-a9a9-a402109934eb" />

denarch22 committed:
Ignore .idea directory and clean up project files

Updated .gitignore to exclude the .idea directory and removed .idea/misc.xml and .idea/workspace.xml from the repository. Also made a minor formatting change in src/utils/storage.py for improved code readability.

cgalvist committed:
refactor: deleted .idea folder
